include/libcyphal/transport/errors.hpp

```diff
@@ -13,34 +13,43 @@ namespace libcyphal
 namespace transport
 {
 
-struct StateError
+struct StateError final
 {};
 
-struct AnonymousError
+struct AnonymousError final
 {};
 
-struct ArgumentError
+struct ArgumentError final
 {};
 
-struct MemoryError
+struct MemoryError final
 {};
 
-struct CapacityError
+struct CapacityError final
 {};
 
-struct PlatformError
+struct PlatformError final
 {
     std::uint32_t code;
 };
 
+struct SessionAlreadyExistsError final
```

Comment:
```suggestion
struct AlreadyExistsError final
```

This way, the error type will be reusable and whether it's related to a session or anything else will be evident from context.